test/app/plugins/delete/delete.test.js

```diff
@@ -65,196 +65,202 @@ function confirmDelete(sidekick) {
+          // preview delete permission is granted
+          preview: {
+            status: 200,
+            permissions: ['read', 'write', 'delete'],
+          },
+        }, null, statusUrl,
+        );
+
+      sidekick = sidekickTest.createSidekick();
+      await sidekickTest.awaitEnvSwitcher();
+
+      await clickDeletePlugin(sidekick);
+
+      expect(showModalSpy.calledWithMatch({ type: MODALS.DELETE })).to.be.true;
+
+      confirmDelete(sidekick);
+
+      await waitUntil(() => deleteStub.calledOnce);
+
+      waitUntil(() => showToastSpy.calledOnce);
+      sidekickTest.clickToastClose();
+
+      expect(deleteStub.calledOnce).to.be.true;
+      expect(reloadPageStub.calledOnce).to.be.false;
+      expect(sidekickTest.rumStub.calledWith('click', {
+        source: 'sidekick',
+        target: 'deleted',
+      })).to.be.true;
+    });
+
+    it('allows authenticated user to delete even if source file still exists', async () => {
+      sidekickTest
+        .mockFetchStatusSuccess(false, {
+          webPath: contentType === HelixMockContentType.DOC ? '/' : '/placeholder.json',
+          edit: {
+            status: 200,
+          },
+          // preview delete permission is granted
+          preview: {
+            status: 200,
+            sourceLocation: 'gdrive:drive-id',
+            permissions: ['read', 'write', 'delete'],
+          },
+          // user authenticated
+          profile: {
+            email: 'foo@example.com',
+            name: 'Peter Parker',
+          },
+        }, null, statusUrl);
+
+      sidekick = sidekickTest.createSidekick();
+      await sidekickTest.awaitEnvSwitcher();
+
+      await clickDeletePlugin(sidekick);
+
+      expect(showModalSpy.calledWithMatch({ type: MODALS.DELETE })).to.be.true;
+
+      confirmDelete(sidekick);
+
+      await waitUntil(() => deleteStub.calledOnce);
+      sidekickTest.clickToastClose();
+      expect(sidekickTest.rumStub.calledWith('click', {
+        source: 'sidekick',
+        target: 'deleted',
+      })).to.be.true;
+    });
+
+    it('handles server failure', async () => {
+      sidekickTest
+        .mockFetchStatusSuccess(false, {
+          // preview delete permission is granted
+          preview: {
+            status: 200,
+            sourceLocation: 'gdrive:drive-id',
+            permissions: ['read', 'write', 'delete'],
+          },
+        }, null, statusUrl);
+
+      sidekick = sidekickTest.createSidekick();
+      await sidekickTest.awaitEnvSwitcher();
+
+      deleteStub.resolves(false);
+
+      await clickDeletePlugin(sidekick);
+
+      expect(showModalSpy.calledWithMatch({ type: MODALS.DELETE })).to.be.true;
+
+      confirmDelete(sidekick);
+
+      await waitUntil(() => deleteStub.calledOnce);
+      expect(deleteStub.calledOnce);
+    });
+  });
+
+  await runTests(HelixMockContentType.DOC);
+  await aTimeout(5000);
+  await runTests(HelixMockContentType.SHEET);
```

Comment:
Refactored tests to prevent concurrency issues while waiting for the toast timeouts.